pkg/api/nft_handlers.go

```diff
@@ -78,3 +79,28 @@ func (h Handler) GetNftCollections(ctx context.Context, params oas.GetNftCollect
 	}
 	return &collectionsRes, nil
 }
+
+func (h Handler) GetNftCollection(ctx context.Context, params oas.GetNftCollectionParams) (oas.GetNftCollectionRes, error) {
+	account, err := tongo.ParseAccountID(params.AccountID)
+	if err != nil {
+		return &oas.BadRequest{Error: err.Error()}, nil
+	}
+
+	collection, err := h.storage.GetNftCollectionByCollectionAddress(ctx, account)
+	if errors.Is(err, core.ErrEntityNotFound) {
+		return &oas.NotFound{Error: err.Error()}, nil
+	}
+	if err != nil {
+		return &oas.InternalError{Error: err.Error()}, nil
+	}
+	col := convertNftCollection(collection)
+	if collection.Metadata != nil {
+		var metadata oas.OptNftCollectionMetadata
```

Comment:
would it make sense to handle the metadata assignment inside `convertNftCollection`?

Reply:
Great point 👌

Reply:
I'm going to change it